src/js/stores/draganddrop.js

```diff
@@ -169,7 +207,7 @@ define(function (require, exports, module) {
                 }.bind(this));
             }
 
-            if ((potentialDropTarget && potentialDropTarget.validate(this._dragTarget, point, potentialDropTarget.b))) {
+            if ((potentialDropTarget && potentialDropTarget.validate(this._dragTarget, point, this._currentBounds))) {
```

Comment:
Initially, I had done the cache bounding rects thing because it seemed to be a clear win speed-wise. However, those bounding rects can change pretty easily (e.g. the layers panel scrolls, layers get reordered). If I calculated them all up front (expensive and that goes right into startup time), there'd be lots of cases where I'd need to invalidate those (because not only do absolute positions change, but relative positions can change), and once invalidated, recalculation is expensive. It seemed overall that calling it once here would minimize the overall number of calls.

I could cache it for the length of the dragging session (and invalidate if there is a scroll event)

Seems like 2. is the biggest target here